packages/@aws-cdk/aws-rds/lib/proxy.ts

```diff
@@ -408,20 +408,30 @@ export class DatabaseProxy extends cdk.Resource
     this.endpoint = this.resource.attrEndpoint;
 
     let dbInstanceIdentifiers: string[] | undefined;
-    if (bindResult.dbClusters) {
```

Comment:
Please add a unit test for the case that was previously wrong and is fixed in this PR.